envoy/network/socket.h

```diff
@@ -112,6 +112,12 @@ class ConnectionInfoProvider {
    * @return roundTripTime of the connection
    */
   virtual const absl::optional<std::chrono::milliseconds>& roundTripTime() const PURE;
+
+  /**
+   * @return absl::string_view the upstream transport failure reason,
+   *         e.g. certificate validation failed.
+   */
+  virtual absl::string_view downstreamTransportFailureReason() const PURE;
```

Comment:
No, there is no addictional copy if you `std::string, initializing std::string` as I mentioned above(last paragraph)
Also, see [go/totw/101](https://goto.google.com/totw/101)

Since you already stored `downstream_transport_failure_reason_` as string, you can just return std::string. ( neither const string& nor absl::string_view)

So, returning `absl::string_view` doesn't save anything.

Reply:
Oh I see, thanks so much! It may make sense to replace others.

Reply:
Also, I do see lots of place returning `absl::string_view` in Envoy code base. I think it is fine to keep as it is.
I just want to mention this for your reference.

Reply:
Good to know this C++17 stuff 😄, thanks!